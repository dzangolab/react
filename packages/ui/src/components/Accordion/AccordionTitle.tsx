import React from "react";

type Properties = {
  title: string;
  icon: string;
  index: number;
  isActive: boolean;
  activeIcon?: string;
  inactiveIcon?: string;
  handleClick: (eventKey: number) => void;
};

const AccordionTitle: React.FC<Properties> = ({
  title,
  index,
  icon,
  isActive,
  handleClick,
  activeIcon,
  inactiveIcon,
}) => {
  return (
    <button onClick={() => handleClick(index)} type="button">
      {icon ? <img src={icon} alt={title} /> : null}
      <span>{title}</span>
      {activeIcon && inactiveIcon ? (
        <img src={isActive ? activeIcon : inactiveIcon} />
      ) : null}
    </button>
  );
};

export default AccordionTitle;
