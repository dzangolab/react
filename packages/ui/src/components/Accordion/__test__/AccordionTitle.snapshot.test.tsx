import React from "react";
import renderer, { create } from "react-test-renderer";
import { expect, test } from "vitest";

import AccordionTitle from "../AccordionTitle";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = create(
    <AccordionTitle
      title="accordion title"
      index={1}
      icon="icon1.jpg"
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      handleClick={() => {}}
    />
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
