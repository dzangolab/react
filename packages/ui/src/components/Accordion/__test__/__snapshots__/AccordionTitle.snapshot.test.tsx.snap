// Vitest Snapshot v1

exports[`Component matches snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  <img
    alt="accordion title"
    src="icon1.jpg"
  />
  <span>
    accordion title
  </span>
  <img
    src="arrow-up.jpg"
  />
</button>
`;
